thirdeye/thirdeye-frontend/app/pods/rootcause/controller.js

```diff
@@ -137,7 +138,10 @@ export default Ember.Controller.extend({
       timeseriesMode: 'split'
     });
 
-    Ember.run.later(this, this._onCheckSessionTimer, ROOTCAUSE_SESSION_TIMER_INTERVAL);
+    // This is a flag for the acceptance test for rootcause to prevent it from timing out because of this run loop
+    if (config.environment !== 'test') {
+      Ember.run.later(this, this._onCheckSessionTimer, this.get('rootcauseTimerInterval'));
```

Comment:
What's the intent of this run loop? Can the intended action be done in a more deterministic way rather than what is essentially a `setTimeout` operation?

Reply:
The run loop is intended to fetch the latest state of the session every 300s. I don't know if there is another way of doing this other than this. Do you have any suggestions on a better way?